pkg/operator/config_reloader.go

```diff
@@ -249,4 +251,43 @@ func CreateConfigReloader(name string, options ...ReloaderOption) v1.Container {
 			},
 		},
 	}
+
+	if !configReloader.runOnce && configReloader.config.EnableProbes {
+		c = addProbes(c)
+	}
+
+	return c
+}
+
+func addProbes(c v1.Container) v1.Container {
+	livenessProbeHandler := v1.ProbeHandler{
+		HTTPGet: &v1.HTTPGetAction{
+			Path: path.Clean("/healthz"),
+			Port: intstr.FromInt(configReloaderPort),
+		},
+	}
+
+	readinessProbeHandler := v1.ProbeHandler{
+		HTTPGet: &v1.HTTPGetAction{
+			Path: path.Clean("/healthz"),
+			Port: intstr.FromInt(configReloaderPort),
+		},
+	}
+
+	livenessProbe := &v1.Probe{
+		ProbeHandler:     livenessProbeHandler,
+		TimeoutSeconds:   3,
+		FailureThreshold: 10,
+	}
+
+	readinessProbe := &v1.Probe{
+		ProbeHandler:     readinessProbeHandler,
+		TimeoutSeconds:   3,
+		PeriodSeconds:    5,
+		FailureThreshold: 10,
+	}
+
+	c.LivenessProbe = livenessProbe
+	c.ReadinessProbe = readinessProbe
+	return c
```

Comment:
I think that we can rely on the default  probe settings (e.g. 3 retries at 10 seconds interval before declaring a failure, 1s time-out).

```suggestion
	probe := &v1.Probe{
		ProbeHandler:     v1.ProbeHandler{
			HTTPGet: &v1.HTTPGetAction{
				Path: path.Clean("/healthz"),
				Port: intstr.FromInt(configReloaderPort),
		},
	}

	c.LivenessProbe = probe
	c.ReadinessProbe = probe
	return c
```

Reply:
Clean Code! Done